tests/tools/traffic-replay/SSLReplay.py

```diff
@@ -0,0 +1,213 @@
+#  limitations under the License.
+
+import http.client
+import socket, ssl, pprint
+import gevent
+import requests
+import os
+#import threading
+import sys
+from multiprocessing import current_process
+import sessionvalidation.sessionvalidation as sv
+import lib.result as result
+import extractHeader
+from gevent import monkey, sleep
+from threading import Thread
+import mainProcess
+import json
+import extractHeader
+import time
+import Config
+bSTOP = False
+
+class ProxyHTTPSConnection(http.client.HTTPSConnection):
+        "This class allows communication via SSL."
+
+        default_port = http.client.HTTPS_PORT
+
+        # XXX Should key_file and cert_file be deprecated in favour of context?
+
+        def __init__(self, host, port=None, key_file=None, cert_file=None,
+                     timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
+                     source_address=None, *, context=None,
+                     check_hostname=None,server_name = None):
+            #http.client.HTTPSConnection.__init__(self)
+            super().__init__(host, port, key_file,cert_file, timeout, source_address,context=context,check_hostname=check_hostname)
+            '''
+            self.key_file = key_file
+            self.cert_file = cert_file
+            if context is None:
+                context = ssl._create_default_https_context()
+            will_verify = context.verify_mode != ssl.CERT_NONE
+            if check_hostname is None:
+                check_hostname = context.check_hostname
+            if check_hostname and not will_verify:
+                raise ValueError("check_hostname needs a SSL context with "
+                                 "either CERT_OPTIONAL or CERT_REQUIRED")
+            if key_file or cert_file:
+                context.load_cert_chain(cert_file, key_file)
+            self._context = context
+            self._check_hostname = check_hostname
+            '''
+            self.server_name = server_name
+
+        def connect(self):
+            "Connect to a host on a given (SSL) port."
+            http.client.HTTPConnection.connect(self)
+
+            if self._tunnel_host:
+                server_hostname = self._tunnel_host
+            else:
+                server_hostname = self.server_name
+            self.sock = self._context.wrap_socket(self.sock,
+                                                do_handshake_on_connect=True,
+                                                server_side=False,
+                                                server_hostname=server_hostname)
+            if not self._context.check_hostname and self._check_hostname:
+                try:
+                    ssl.match_hostname(self.sock.getpeercert(), server_hostname)
+                except Exception:
+                    self.sock.shutdown(socket.SHUT_RDWR)
+                    self.sock.close()
+                    raise
+
+
+def txn_replay(session_filename, txn, proxy, result_queue, request_session):
```

Comment:
Why is this code duplicated? It should be moved to sub functions, if not a method on an inherited base class.